src/Database/LSMTree/Internal/IndexCompactAcc.hs

```diff
@@ -126,26 +122,6 @@ newPinnedMVec64 lenWords = do
     setByteArray mba 0 lenWords (0 :: Word64)
     return (VUM.MV_Word64 (VPM.MVector 0 lenWords mba))
 
--- | Min\/max key-info for pages
-data Append =
-    -- | One or more keys are in this page, and their values fit within a single
-    -- page.
-    AppendSinglePage SerialisedKey SerialisedKey
-    -- | There is only one key in this page, and it's value does not fit within
-    -- a single page.
-  | AppendMultiPage SerialisedKey Word32 -- ^ Number of overflow pages
-
-instance NFData Append where
-  rnf (AppendSinglePage kmin kmax)  = rnf kmin `seq` rnf kmax
-  rnf (AppendMultiPage k nOverflow) = rnf k `seq` rnf nOverflow
-
--- | Append a new page entry to a mutable compact index.
---
--- INVARIANTS: see [construction invariants](#construction-invariants).
```

Comment:
This invariant goes missing. Are we going to have preconditions/invariants on the index class that capture the requirements for both the compact and ordinary index?

Reply:
Yes, it is the intention to have the sortedness invariant, which is the one invariant currently stated for the compact index, as a general invariant for all indexes and to have corresponding documentation along with the index type class declaration. That said, it’s surely a good idea to refer to the usage restrictions for appending also here. I’ve added a corresponding statement that works in the present situation and will continue to work when the index type class is introduced. The commit introducing this change is 0bf6c30ec10fa89c0fcc0489f2b75dba6a4e4edd.